### DISCUSSION TITLE
How get information about an already signed subscription? (Android)

### DISCUSSION THREAD

**Discussion by yuriydrobniy** [hyochan/react-native-iap#1900] [Q&A]
Hey guys!
Could someone help me with question. 
How can I get information about an already signed subscription? 
My goal is to get the price of this subscription. And the problem is that we only have the `productId` to determine which subscription is purchased. But in Google Play Billing Library V5 we have multiple offers with unique `offerToken`s under the same `productId`.

Im using react-native-iap: 9.0.0

**Comment by andresesfm** [comment]
@yuriydrobniy Do you mean that you want to know what `offerToken` you used to purchase a given `productId`? If so, have you considered using `developerPayload`?

**Reply by professorkolik** [reply]
@andresesfm any other ways in your mind?

Since `developerPayload` is not pretty accurate, of course it's better than storing it in DB, but anyway.

As far as I see it doesn't exist in BillingLibrary itself, correct me if I'm wrong.

**Reply by andresesfm** [reply]
@professorkolik can you please elaborate on `developerPayload` not being accurate? It's a string whatever you set is what you get back. I'm curious, let me know

**Reply by professorkolik** [reply]
@andresesfm First of all as far I can see `developPayload` is deprecated from BillingClient v3 according to this [reference](https://developer.android.com/google/play/billing/developer-payload)
Second I feel it's not related to `developer` (responsibility vice) to pass offerToken, which I would expect should come in Purchase data. It's actually strange that Google doesn't provide it out of the box (as far as I can see the docs)

And the 3d even when we pass developerPayload in finishTransaction it doesn't exist then in `availablePurchase` (we acknowledge purchase on BE with help of Real-time developer notifications)

**Reply by andresesfm** [reply]
I can't think of a better option than storing it in your backend. even developerPayload was more of a hack
